scipy/special/tests/test_zeta.py

```diff
@@ -47,3 +49,89 @@ def test_riemann_zeta_avoid_overflow():
     s = -260.00000000001
     desired = -5.6966307844402683127e+297  # Computed with Mpmath
     assert_allclose(sc.zeta(s), desired, atol=0, rtol=5e-14)
+
+
+@pytest.mark.parametrize(
+    "z, desired",
+    # Reference cases taken from mpmath with the script:
+    # import numpy as np
+    # import scipy.stats as stats
+
+    # from mpmath import mp
+
+    # # seed = np.random.SeedSequence().entropy
+    # seed = 154689806791763421822480125722191067828
+    # rng = np.random.default_rng(seed)
+
+    # # A small point in each quadrant outside of the critical strip
+    # cases = []
+    # for x_sign, y_sign in [1, 1], [1, -1], [-1, 1], [-1, -1]:
+    #     x = x_sign * rng.uniform(2, 8)
+    #     y = y_sign * rng.uniform(2, 8)
+    #     z = x + y*1j
+    #     reference = complex(mp.zeta(z))
+    #     cases.append((z, reference))
+
+    # # Moderately large imaginary part in each quadrant outside of critical strip
+    # for x_sign, y_sign in [1, 1], [1, -1], [-1, 1], [-1, -1]:
+    #     x = x_sign * rng.uniform(2, 8)
+    #     y = y_sign * rng.uniform(50, 80)
+    #     z = x + y*1j
+    #     reference = complex(mp.zeta(z))
+    #     cases.append((z, reference))
+
+    # # points in critical strip
```

Comment:
More precisely, we evaluate as though it's in the critical strip (possibly applying reflection formula)?

Reply:
Ah, right, that's correct. I'll change the comment.

Reply:
Actually points with real part less than 0 - even greater than -1 - don't get reflected into the critical strip. So you could either adjust the comment to "point in and near critical strip" (if there's a reason why real part between -1 and 0 is concerning) or just limit the test to the critical strip.

Reply:
I'll limit the test to the critical strip. Just an oversight, I didn't put enough thought into this.